plugins/router/f5/f5.go

```diff
@@ -0,0 +1,555 @@
+	}
+
+	glog.V(4).Infof("Adding pool member %s to pool %s.",
+		member, poolName)
+
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s/members",
+		f5.host, poolName)
+
+	payload := struct {
+		Member string `json:"name"`
+	}{
+		Member: member,
+	}
+
+	err = f5.post(url, payload, nil)
+	if err != nil {
+		return err
+	}
+
+	members, err := f5.GetPoolMembers(poolName)
+	if err != nil {
+		return err
+	}
+
+	members[member] = true
+
+	glog.V(4).Infof("Added pool member %s to pool %s.",
+		member, poolName)
+
+	return nil
+}
+
+// Delete member from the pool named poolName on F5, and update
+// f5.poolMembers[poolName].
+func (f5 *f5LTM) DeletePoolMember(poolName string, member string) error {
+	// The invocation of f5.PoolHasMember has the side effect that it will
+	// initialise f5.poolMembers[poolName], which is used below, if necessary.
+	hasMember, err := f5.PoolHasMember(poolName, member)
+	if err != nil {
+		return err
+	}
+	if !hasMember {
+		return fmt.Errorf("Pool %s does not have member %s.\n", poolName, member)
+	}
+
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s/members/%s",
+		f5.host, poolName, member)
+
+	err = f5.delete(url, nil)
+	if err != nil {
+		return err
+	}
+
+	f5.poolMembers[poolName][member] = false
+
+	glog.V(4).Infof("Pool member %s deleted from pool %s.", member, poolName)
+
+	return nil
+}
+
+// Add a rule to the policy on F5 to route requests to the given hostname to
+// the pool named by poolName.
+func (f5 *f5LTM) AddPoolRoute(poolName string, hostname string) error {
+	success := false
+
+	rulesUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/policy/%s/rules",
+		f5.host, policy)
+
+	rulesPayload := struct {
+		Name string `json:"name"`
+	}{
+		Name: hostname,
+	}
+
+	err := f5.post(rulesUrl, rulesPayload, nil)
+	if err != nil {
+		if err.(F5Error).httpStatusCode == 409 {
+			glog.V(4).Infof("Warning: Rule already exists for hostname %s; continuing with initialization in case the existing rule is only partially initialized...", hostname)
+		} else {
+			return err
+		}
+	}
+
+	// If adding the condition or action to the rule fails later on,
+	// delete the rule.
+	defer func() {
+		if success != true {
+			f5.DeletePoolRoute(hostname)
+		}
+	}()
+
+	conditionsUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/policy/%s/rules/%s/conditions",
+		f5.host, policy, hostname)
+
+	conditionsPayload := struct {
+		Name            string   `json:"name"`
+		CaseInsensitive bool     `json:"caseInsensitive"`
+		External        bool     `json:"external"`
+		HttpHost        bool     `json:"httpHost"`
+		Index           int      `json:"index"`
+		Equals          bool     `json:"equals"`
+		Present         bool     `json:"present"`
+		Remote          bool     `json:"remote"`
+		Request         bool     `json:"request"`
+		Values          []string `json:"values"`
+	}{
+		Name:            "0",
+		CaseInsensitive: true,
+		External:        true,
+		HttpHost:        true,
```

Comment:
It may be my F5 ignorance, but how does this support tcp passthrough routes and the other route types like edge terminated and reencrypt?